### PR TITLE
BAEL-2400: Adds example for currying

### PR DESCRIPTION
**Pull request by hauer-io**

### CODE REVIEW

File: patterns/design-patterns/src/test/java/com/baeldung/currying/LetterUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import static com.baeldung.currying.Letter.SIMPLE_LETTER_CREATOR;
+import static org.junit.Assert.assertEquals;
+
+public class LetterUnitTest {
+    private final static String BODY = "BODY";
+    private final static String SALUTATION = "SALUTATION";
+    private final static String RETURNING_ADDRESS = "RETURNING ADDRESS";
+    private final static String INSIDE_ADDRESS = "INSIDE ADDRESS";
+    private final static LocalDate DATE_OF_LETTER = LocalDate.of(2013, 3, 1);
+    private final static String CLOSING = "CLOSING";
```
Comment (ashleyfrieze) [inline patterns/design-patterns/src/test/java/com/baeldung/currying/LetterUnitTest.java]: These should be `static final` not `final static`

File: patterns/design-patterns/src/main/java/com/baeldung/currying/Letter.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    Letter(String salutation, String body) {
+        this(null, null, LocalDate.now(), salutation, body, null);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
```
Comment (ashleyfrieze) [inline patterns/design-patterns/src/main/java/com/baeldung/currying/Letter.java]: These should have `{}` around the body of the if - to be in line with most convention.

File: patterns/design-patterns/src/test/java/com/baeldung/currying/LetterUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Test
+    public void whenStaticSimpleCurriedFunctionIsCalled_thenaSimpleLetterIsCreated() {
+        assertEquals(SIMPLE_LETTER, SIMPLE_CURRIED_LETTER_CREATOR.apply(SALUTATION).apply(BODY));
+    }
+
+    @Test
+    public void whenStaticCurriedFunctionIsCalled_thenaLetterIsCreated() {
+        assertEquals(LETTER, LETTER_CREATOR
+                                    .apply(RETURNING_ADDRESS)
```
Comment (KevinGilmore) [inline patterns/design-patterns/src/test/java/com/baeldung/currying/LetterUnitTest.java]: use standard indentation for continuation lines

File: patterns/design-patterns/src/test/java/com/baeldung/currying/LetterUnitTest.java (hunk 1 above)
Comment (hauer-io) [reply patterns/design-patterns/src/test/java/com/baeldung/currying/LetterUnitTest.java]: :heavy_check_mark:

File: patterns/design-patterns/src/main/java/com/baeldung/currying/Letter.java (hunk 2 above)
Comment (hauer-io) [reply patterns/design-patterns/src/main/java/com/baeldung/currying/Letter.java]: :check

Comment (hauer-io) [reply patterns/design-patterns/src/main/java/com/baeldung/currying/Letter.java]: :heavy_check_mark:

File: patterns/design-patterns/src/test/java/com/baeldung/currying/LetterUnitTest.java (hunk 3 above)
Comment (hauer-io) [reply patterns/design-patterns/src/test/java/com/baeldung/currying/LetterUnitTest.java]: :heavy_check_mark:

**Pull request merged**